storage/table_test.go

```diff
@@ -0,0 +1,298 @@
+
+	tn := AzureTable(randTable())
+
+	err := cli.CreateTable(tn)
+	c.Assert(err, chk.IsNil)
+	defer cli.DeleteTable(tn)
+
+	ce := &CustomEntity{Name: "Darth", Surname: "Skywalker", SomeDate: time.Now(), Number: 60, PKey: "pkey", RKey: "100"}
+	c.Assert(cli.InsertOrReplaceEntity(tn, ce), chk.IsNil)
+
+	ce.SetRowKey("200")
+	c.Assert(cli.InsertOrReplaceEntity(tn, ce), chk.IsNil)
+
+	entries, _, err := cli.QueryTableEntities(tn, nil, reflect.TypeOf(ce), 10, "RowKey eq '200'")
+	c.Assert(err, chk.IsNil)
+
+	c.Assert(len(*entries), chk.Equals, 1)
+
+	c.Assert(ce.RowKey(), chk.Equals, (*entries)[0].RowKey())
+}
+
+func (s *StorageBlobSuite) Test_InsertAndDeleteEntities(c *chk.C) {
+	cli := getTableClient(c)
+
+	tn := AzureTable(randTable())
+
+	err := cli.CreateTable(tn)
+	c.Assert(err, chk.IsNil)
+	defer cli.DeleteTable(tn)
+
+	ce := &CustomEntity{Name: "Test", Surname: "Test2", SomeDate: time.Now(), Number: 0, PKey: "pkey", RKey: "r01"}
+	c.Assert(cli.InsertOrReplaceEntity(tn, ce), chk.IsNil)
+
+	ce.Number = 1
+	ce.SetRowKey("r02")
+	c.Assert(cli.InsertOrReplaceEntity(tn, ce), chk.IsNil)
+
+	entries, _, err := cli.QueryTableEntities(tn, nil, reflect.TypeOf(ce), 10, "Number eq 1")
+	c.Assert(err, chk.IsNil)
+
+	c.Assert(len(*entries), chk.Equals, 1)
+
+	testEquality(c, ce, (*entries)[0].(*CustomEntity))
+
+	c.Assert(cli.DeleteEntityWithoutCheck(tn, (*entries)[0]), chk.IsNil)
+
+	entries, _, err = cli.QueryTableEntities(tn, nil, reflect.TypeOf(ce), 10, "")
+	c.Assert(err, chk.IsNil)
+
+	// only 1 entry must be present
+	c.Assert(len(*entries), chk.Equals, 1)
+}
+
+func (s *StorageBlobSuite) Test_ContinuationToken(c *chk.C) {
+	cli := getTableClient(c)
+
+	tn := AzureTable(randTable())
+
+	err := cli.CreateTable(tn)
+	c.Assert(err, chk.IsNil)
+	defer cli.DeleteTable(tn)
+
+	var ce *CustomEntity
+	for i := 0; i < 5; i++ {
+		ce = &CustomEntity{Name: "Test", Surname: "Test2", SomeDate: time.Now(), Number: i, PKey: "pkey", RKey: fmt.Sprintf("r%d", i)}
+		c.Assert(cli.InsertOrReplaceEntity(tn, ce), chk.IsNil)
+	}
+
+	// retrieve using top = 2. Should return 2 entries, 2 entries and finally
+	// 1 entry
+	entries, contToken, err := cli.QueryTableEntities(tn, nil, reflect.TypeOf(ce), 2, "")
+	c.Assert(err, chk.IsNil)
+	c.Assert(len(*entries), chk.Equals, 2)
+	c.Assert(contToken, chk.NotNil)
+
+	entries, contToken, err = cli.QueryTableEntities(tn, contToken, reflect.TypeOf(ce), 2, "")
+	c.Assert(err, chk.IsNil)
+	c.Assert(len(*entries), chk.Equals, 2)
+	c.Assert(contToken, chk.NotNil)
+
+	entries, contToken, err = cli.QueryTableEntities(tn, contToken, reflect.TypeOf(ce), 2, "")
+	c.Assert(err, chk.IsNil)
+	c.Assert(len(*entries), chk.Equals, 1)
+	c.Assert(contToken, chk.IsNil)
+}
+
+func randTable() string {
+	const alphanum = "abcdefghijklmnopqrstuvwxyz"
+	var bytes = make([]byte, 32)
+	rand.Read(bytes)
+	for i, b := range bytes {
+		bytes[i] = alphanum[b%byte(len(alphanum))]
+	}
+	return string(bytes)
+}
+
+func testEquality(c *chk.C, c1, c2 *CustomEntity) {
```

Comment:
hi @MindFlavor just started to take a look at this PR. You can use `c.Assert(a, DeepEquals, b)` to eliminate need for this method.